manifests/rule/remote.pp

```diff
@@ -160,7 +166,7 @@
   Optional[String]                                 $stream_driver                        = undef,
   Integer[0]                                       $stream_driver_mode                   = 1,
   String                                           $stream_driver_auth_mode              = 'x509/name',
-  String                                           $stream_driver_permitted_peers        = "*.${::domain}",
+  String                                           $stream_driver_permitted_peers        = 'LOGSERVERNAME',
```

Comment:
It seems like we should change this to `Optional[Variant[String[1],Array[String[1]]]]`.

This would allow the logic to not have any magic due to a special variable that could potentially be a valid value in someone's environment.

I do think that the logic below makes sense but, in theory, we should allow people to specify an array of permitted peers (since it's plural) and the logic would change to acting on whether or not the parameter is defined.